### PR TITLE
Build: Bump com.adobe.testing:s3mock-junit5 from 2.11.0 to 2.17.0

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: aws/src/test/java/org/apache/iceberg/aws/s3/TestS3FileIO.java [hunk 1]
```diff
@@ -447,7 +448,7 @@ private void createRandomObjects(String prefix, int count) {
   private void createBucket(String bucketName) {
     try {
       s3.get().createBucket(CreateBucketRequest.builder().bucket(bucketName).build());
-    } catch (BucketAlreadyExistsException e) {
+    } catch (BucketAlreadyExistsException | BucketAlreadyOwnedByYouException e) {
```
Comment (Fokko) [inline aws/src/test/java/org/apache/iceberg/aws/s3/TestS3FileIO.java:451]: @nastra this seems unrelated?

**Review by Fokko — APPROVED** [review]

File: aws/src/test/java/org/apache/iceberg/aws/s3/TestS3FileIO.java (hunk 1 above)
Comment (nastra) [reply aws/src/test/java/org/apache/iceberg/aws/s3/TestS3FileIO.java:451]: this is actually required to make the tests pass

Comment (nastra) [reply aws/src/test/java/org/apache/iceberg/aws/s3/TestS3FileIO.java:451]: the newer version of the API throws a new exception which the tests most likely just want to ignore

Comment (Fokko) [reply aws/src/test/java/org/apache/iceberg/aws/s3/TestS3FileIO.java:451]: Ah I see, thanks 👍

**Comment by nastra** [comment]
thanks for the review @Fokko
